pages/company/visualizations/comapny_associated_activity.py

```diff
@@ -0,0 +1,256 @@
+    cache = cm()
+    df = cache.grabm(func=cq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=cq, repos=repolist)
+
+    # get date value for first contribution
+    df["created"] = pd.to_datetime(df["created"], utc=True)
+    df = df.sort_values(by="created", axis=0, ascending=True)
+    base = df.iloc[0]["created"]
+
+    date_picker = (
+        dcc.DatePickerRange(
+            id=f"{PAGE}-date-picker-range-{VIZ_ID}",
+            min_date_allowed=base,
+            max_date_allowed=dt.date.today(),
+            clearable=True,
+        ),
+    )
+    return date_picker
+
+
+# callback for Company Affiliation by Github Account Info graph
+@callback(
+    Output(VIZ_ID, "figure"),
+    [
+        Input("repo-choices", "data"),
+        Input(f"{PAGE}-{paramter_2}-{VIZ_ID}", "value"),
+        Input(f"{PAGE}-{paramter_1}-{VIZ_ID}", "value"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "start_date"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "end_date"),
+    ],
+    background=True,
+    prevent_initial_call=True,
+)
+def compay_associated_activity_graph(repolist, checks, num, start_date, end_date):
+
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=cq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=cq, repos=repolist)
+
+    start = time.perf_counter()
+    logging.debug(f"{VIZ_ID}- START")
+
+    # test if there is data
+    if df.empty:
+        logging.debug(f"{VIZ_ID} - NO DATA AVAILABLE")
+        return nodata_graph
+
+    # function for all data pre processing, COULD HAVE ADDITIONAL INPUTS AND OUTPUTS
+    df = process_data(df, checks, num, start_date, end_date)
+
+    fig = create_figure(df)
+
+    logging.debug(f"{VIZ_ID} - END - {time.perf_counter() - start}")
+    return fig
+
+
+def process_data(df: pd.DataFrame, checks, num, start_date, end_date):
+    """Implement your custom data-processing logic in this function.
+    The output of this function is the data you intend to create a visualization with,
+    requiring no further processing."""
+
+    # convert to datetime objects rather than strings
+    df["created"] = pd.to_datetime(df["created"], utc=True)
+
+    # order values chronologically by COLUMN_TO_SORT_BY date
+    df = df.sort_values(by="created", axis=0, ascending=True)
+
+    # filter values based on date picker
+    if start_date is not None:
+        df = df[df.created >= start_date]
+    if end_date is not None:
+        df = df[df.created <= end_date]
+
+    # creates list of emails for each contribution and flattens list result
+    emails = df.email_list.str.split(" , ").explode("email_list").tolist()
+
+    # remove any entries not in email format
+    emails = [x for x in emails if "@" in x]
+
+    # creates list of email domains from the emails list
+    email_domains = [x[x.rindex("@") + 1 :] for x in emails]
```

Comment:
May want to use an email-parsing regular expression in the future. there can be some funky non-standard emails out there.